crates/storage/src/rocksdb.rs

```diff
@@ -0,0 +1,149 @@
+use crate::Database;
+use anyhow::Result;
+use jmt::{
+    storage::{LeafNode, Node, NodeBatch, NodeKey, TreeReader, TreeWriter},
+    KeyHash, OwnedValue, Version,
+};
+use prism_common::tree::Digest;
+use prism_errors::DatabaseError;
+use rocksdb::{DBWithThreadMode, MultiThreaded, Options, DB};
+
+type RocksDB = DBWithThreadMode<MultiThreaded>;
+
+pub struct RocksDBConnection {
+    connection: RocksDB,
+    path: String,
+}
+
+impl RocksDBConnection {
+    pub fn new(path: &str) -> Result<RocksDBConnection> {
+        let db = DB::open_default(path)?;
+
+        Ok(Self {
+            connection: db,
+            path: path.to_string(),
+        })
+    }
+}
+
+impl Database for RocksDBConnection {
+    fn get_commitment(&self, epoch: &u64) -> anyhow::Result<Digest> {
+        let key = format!("commitments:epoch_{}", epoch);
+        let raw_bytes = self.connection.get(key.as_bytes())?.ok_or_else(|| {
+            DatabaseError::NotFoundError(format!("commitment from epoch_{}", epoch))
+        })?;
+
+        let value: [u8; 32] = raw_bytes
+            .try_into()
+            .expect("commitment digest should always be 32 bytes");
+
+        Ok(Digest(value))
+    }
+
+    fn set_commitment(&self, epoch: &u64, commitment: &Digest) -> anyhow::Result<()> {
+        Ok(self.connection.put::<&[u8], [u8; 32]>(
+            format!("commitments:epoch_{}", epoch).as_bytes(),
+            commitment.0,
+        )?)
+    }
+
+    fn get_epoch(&self) -> anyhow::Result<u64> {
+        let res = self
+            .connection
+            .get(b"app_state:epoch")?
+            .ok_or_else(|| DatabaseError::NotFoundError("current epoch".to_string()))?;
+
+        Ok(u64::from_be_bytes(res.try_into().unwrap()))
+    }
+
+    fn set_epoch(&self, epoch: &u64) -> anyhow::Result<()> {
+        Ok(self
+            .connection
+            .put(b"app_state:epoch", epoch.to_be_bytes())?)
+    }
+
+    fn flush_database(&self) -> Result<()> {
+        Ok(DB::destroy(&Options::default(), &self.path)?)
+    }
+}
+
+impl TreeWriter for RocksDBConnection {
+    fn write_node_batch(&self, node_batch: &NodeBatch) -> Result<()> {
+        todo!()
```

Comment:
Let's follow this up in a next PR, this one is getting massive because of the refactorings